CMakeLists.txt

```diff
@@ -358,6 +364,45 @@ elseif(UNIX)
     FILES ${PROJECT_SOURCE_DIR}/packaging/linux/common/icons/scalable/notes.svg
     DESTINATION ${CMAKE_INSTALL_DATAROOTDIR}/icons/hicolor/scalable/apps
     RENAME ${APP_ID}.svg)
+
+  install(
+    FILES ${PROJECT_SOURCE_DIR}/LICENSE
+    DESTINATION ${CMAKE_INSTALL_DATAROOTDIR}/licenses/${PROJECT_NAME_LOWERCASE})
+
+  # This works around an annoying issue when packaging with different versions of CMake.
+  # In newer versions (probably > 3.10), the description of the package could get duplicated.
+  # In older versions, the description would not get added at all (only a short summary).
+  # In short, it's a whole mess.
+  set(CPACK_PACKAGE_DESCRIPTION_SUMMARY "")
+
+  # Please keep the character count for every line at 80 (max) - it's a lintian rule.
+  string(
+    CONCAT
+      DEBIAN_PACKAGE_DESCRIPTION
+      "${PROJECT_DESCRIPTION}\n"
+      " Notes is an open-source and cross-platform note-taking app that is both\n"
+      " beautiful and powerful. It is written in C++ using the Qt framework.")
+
+  # Include CPack for creating deb/rpm packages
+  set(CPACK_PACKAGE_NAME ${PROJECT_NAME_LOWERCASE})
+  set(CPACK_PACKAGE_VERSION ${PROJECT_VERSION}${GIT_REV})
+  set(CPACK_PACKAGE_CONTACT "${APP_AUTHOR} <${APP_AUTHOR_EMAIL}>")
+
+  # deb-specific
+  set(CPACK_DEBIAN_FILE_NAME DEB-DEFAULT)
+  set(CPACK_DEBIAN_PACKAGE_SHLIBDEPS ON)
+  set(CPACK_DEBIAN_PACKAGE_GENERATE_SHLIBS ON)
+  set(CPACK_DEBIAN_PACKAGE_GENERATE_SHLIBS_POLICY ">=")
+  set(CPACK_DEBIAN_PACKAGE_HOMEPAGE ${APP_WEBSITE})
+  set(CPACK_DEBIAN_PACKAGE_SECTION "utils")
+  set(CPACK_DEBIAN_PACKAGE_DESCRIPTION ${DEBIAN_PACKAGE_DESCRIPTION})
+
+  # Workaround for the Qt6 .deb package requiring a fixed version of qt6-base-abi
+  if(QT_VERSION_MAJOR EQUAL 6)
+    set(CPACK_DEBIAN_PACKAGE_DEPENDS "qt6-base-abi (>= 6.0)")
```

Comment:
I'm not 100% sure about this change... It might be a dangerous game to be playing, so I'm still in favor of just having different debs for different versions of Ubuntu.

Which, if we think about it, is exactly what Ubuntu does with their own packages.

Reply:
Do the packages have the correct versions without this change?

Reply:
Yes. Although you won't be able to install the Qt 6 package built on Ubuntu 22.04, on Ubuntu 22.10 (and vice-versa).

You also mentioned earlier that you are on Ubuntu 22.04, but can you double-check that you're not on 22.10 instead? As far as I can tell, Qt 6.3.1 is not present on Ubuntu 22.04, only on 22.10.

Reply:
Yes, my VM is on 22.10. Not sure why I thought I was on 22.04.